src/app/ipv/tests/handleJourneyActionRequest.test.ts

```diff
@@ -2,161 +2,178 @@ import { expect } from "chai";
-    } catch (error) {
-      expect(
-        coreBackServiceStub.postJourneyEvent.firstCall,
-      ).to.have.been.calledWith(
-        req,
-        "some-journey-event",
-        req.session.currentPage,
-      );
-      expect(error).to.be.an.instanceOf(Error);
-    }
+    // Arrange
+    const req = createRequest({ body: { journey: "some-journey-event" } });
+    const res = createResponse();
+
+    // Act & Assert
+    await expect(
+      middleware.handleJourneyActionRequest(req, res, next),
+    ).to.be.rejectedWith(Error);
+    expect(
+      coreBackServiceStub.postJourneyEvent,
+    ).to.have.been.calledOnceWithExactly(
+      req,
+      "some-journey-event",
+      req.session.currentPage,
+    );
   });
 
   it("should postJourneyEvent and use ip address from header when not present in session", async function () {
-    const req = {
-      ...testReq,
+    // Arrange
+    const req = createRequest({
       headers: { forwarded: "1.1.1.1" },
-      session: { ...testReq.session, ipAddress: undefined },
-    };
+      session: { ipAddress: undefined },
+    });
+    const res = createResponse();
 
-    try {
-      await middleware.handleJourneyActionRequest(req, res, next);
-    } catch (error) {
-      expect(
-        coreBackServiceStub.postJourneyEvent.firstCall,
-      ).to.have.been.calledWith(req, req.body.journey, req.session.currentPage);
-      expect(error).to.be.an.instanceOf(Error);
-    }
+    // Act & Assert
+    await expect(
+      middleware.handleJourneyActionRequest(req, res, next),
+    ).to.be.rejectedWith(Error);
+    expect(
+      coreBackServiceStub.postJourneyEvent,
+    ).to.have.been.calledOnceWithExactly(
+      req,
+      req.body.journey,
+      req.session.currentPage,
+    );
   });
 
   it("should postJourneyEvent and use ip address from session when present", async function () {
-    const req = {
-      ...testReq,
-      headers: { forwarded: "1.1.1.1" },
-      session: { ...testReq.session, ipAddress: "some-ip-address" },
-    };
+    // Arrange
+    const req = createRequest({ session: { ipAddress: "some-ip-address" } });
+    const res = createResponse();
 
-    try {
-      await middleware.handleJourneyActionRequest(req, res, next);
-    } catch (error) {
-      expect(
-        coreBackServiceStub.postJourneyEvent.firstCall,
-      ).to.have.been.calledWith(req, req.body.journey, req.session.currentPage);
-      expect(error).to.be.an.instanceOf(Error);
-    }
+    // Act & Assert
+    await expect(
+      middleware.handleJourneyActionRequest(req, res, next),
+    ).to.be.rejectedWith(Error);
+    expect(
+      coreBackServiceStub.postJourneyEvent,
+    ).to.have.been.calledOnceWithExactly(
+      req,
+      req.body.journey,
+      req.session.currentPage,
+    );
   });
 
-  it("should call redirect given 'contact' event", async function () {
-    const req = { ...testReq, body: { journey: "contact" } };
+  it("should use session IP if present", async function () {
```

Comment:
Again there doesn't appear to be any testing of the IP address - so what is this test for?